localstack/services/dynamodb/dynamodb_listener.py

```diff
@@ -769,6 +839,76 @@ def dynamodb_get_table_stream_specification(table_name):
         raise e
 
 
+def is_kinesis_stream_exists(stream_arn):
+    # connect to kinesis
+    kinesis = aws_stack.connect_to_service('kinesis')
+    # check if the stream exists in kinesis for the user
+    filtered = list(filter(lambda stream_name: stream_name == stream_arn.split('/', 1)[1],
+        kinesis.list_streams()['StreamNames']))
+    if filtered:
+        return True
+    return False
+
+
+def dynamodb_enable_kinesis_streaming_destination(data, table_def):
+    if (table_def.get('KinesisDataStreamDestinationStatus') in ['DISABLED', 'ENABLE_FAILED', None]):
+        table_def['KinesisDataStreamDestinations'] = table_def.get('KinesisDataStreamDestinations') or []
+        # remove the destination if already present
+        for dest in table_def['KinesisDataStreamDestinations']:
+            if dest['StreamArn'] == data['StreamArn']:
+                table_def['KinesisDataStreamDestinations'].remove(dest)
+        # Add the destination on the top of the stack
+        # so that the active destination is always present at the top
+        table_def['KinesisDataStreamDestinations'].append({
+            'DestinationStatus': 'ACTIVE',
+            'DestinationStatusDescription': 'Stream is active',
+            'StreamArn': data['StreamArn']
+        })
+        table_def['KinesisDataStreamDestinationStatus'] = 'ACTIVE'
+        response = Response()
+        response.status_code = 200
+        response._content = json.dumps({
+            'DestinationStatus': 'ACTIVE',
+            'StreamArn': data['StreamArn'],
+            'TableName': data['TableName']
+        })
+        fix_headers_for_updated_response(response)
+        return response
+
+    return error_response(
+        error_type='ValidationException',
+        message='Table is not in a valid state to enable Kinesis Streaming '
+        'Destination:EnableKinesisStreamingDestination must be DISABLED or ENABLE_FAILED '
+        'to perform ENABLE operation.'
+    )
+
+
+def dynamodb_disable_kinesis_streaming_destination(data, table_def):
+    if table_def.get('KinesisDataStreamDestinations'):
+        if table_def['KinesisDataStreamDestinationStatus'] == 'ACTIVE':
+
+            for dest in table_def['KinesisDataStreamDestinations']:
+                if dest['StreamArn'] == data['StreamArn'] and dest['DestinationStatus'] == 'ACTIVE':
+                    dest['DestinationStatus'] = 'DISABLED'
+                    dest['DestinationStatusDescription'] = 'Stream is disabled',
+
+                    table_def['KinesisDataStreamDestinationStatus'] = 'DISABLED'
+                    response = Response()
```

Comment:
nit: same as above - easier to use `aws_responses.requests_response({...})`

Reply:
Thanks @whummer for all the comments and nit suggestions.